snappy-core/src/main/scala/org/apache/spark/streaming/SnappyStreamingContext.scala

```diff
@@ -0,0 +1,290 @@
+    }
+  }
+
+  private def setInstanceContext(snsc: SnappyStreamingContext): Unit = {
+    ACTIVATION_LOCK.synchronized {
+      instanceContext.set(snsc)
+    }
+  }
+
+  /**
+   * :: Experimental ::
+   *
+   * Get the currently created context, it may be started or not, but never stopped.
+   */
+  @Experimental
+  def getInstance(): Option[SnappyStreamingContext] = {
+    ACTIVATION_LOCK.synchronized {
+      Option(instanceContext.get())
+    }
+  }
+
+
+  /**
+   * :: Experimental ::
+   *
+   * Get the currently active context, if there is one. Active means started but not stopped.
+   */
+  @Experimental
+  def getActive(): Option[SnappyStreamingContext] = {
+    ACTIVATION_LOCK.synchronized {
+      Option(activeContext.get())
+    }
+  }
+
+  /**
+   * :: Experimental ::
+   *
+   * Either return the "active" StreamingContext (that is, started but not stopped), or create a
+   * new StreamingContext that is
+   * @param creatingFunc   Function to create a new StreamingContext
+   */
+  @Experimental
+  def getActiveOrCreate(creatingFunc: () => SnappyStreamingContext): SnappyStreamingContext = {
+    ACTIVATION_LOCK.synchronized {
+      getActive().getOrElse { creatingFunc() }
+    }
+  }
+
+  /**
+   * :: Experimental ::
+   *
+   * Either get the currently active StreamingContext (that is, started but not stopped),
+   * OR recreate a StreamingContext from checkpoint data in the given path. If checkpoint data
+   * does not exist in the provided, then create a new StreamingContext by calling the provided
+   * `creatingFunc`.
+   *
+   * @param checkpointPath Checkpoint directory used in an earlier StreamingContext program
+   * @param creatingFunc   Function to create a new StreamingContext
+   * @param hadoopConf     Optional Hadoop configuration if necessary for reading from the
+   *                       file system
+   * @param createOnError  Optional, whether to create a new StreamingContext if there is an
+   *                       error in reading checkpoint data. By default, an exception will be
+   *                       thrown on error.
+   */
+  @Experimental
+  def getActiveOrCreate(
+      checkpointPath: String,
+      creatingFunc: () => SnappyStreamingContext,
+      hadoopConf: Configuration = SparkHadoopUtil.get.conf,
+      createOnError: Boolean = false
+      ): SnappyStreamingContext = {
+    ACTIVATION_LOCK.synchronized {
+      getActive().getOrElse { getOrCreate(checkpointPath, creatingFunc, hadoopConf, createOnError) }
+    }
+  }
+
+  def getOrCreate(
+      checkpointPath: String,
+      creatingFunc: () => SnappyStreamingContext,
+      hadoopConf: Configuration = SparkHadoopUtil.get.conf,
+      createOnError: Boolean = false
+      ): SnappyStreamingContext = {
+    val checkpointOption = CheckpointReader.read(
+      checkpointPath, new SparkConf(), hadoopConf, createOnError)
+    checkpointOption.map(new SnappyStreamingContext(null, _, null)).getOrElse(creatingFunc())
```

Comment:
1. Throw an exception if a streaming context is already created. 
2. set this newly created streaming context as the current instance